### PR TITLE
Filter the language term description

### PR DESCRIPTION
**Pull request by Screenfeed**
This allows to store new data like the locale fallbacks.
This will be used by https://github.com/polylang/polylang-pro/pull/1573.

### CODE REVIEW

File: admin/admin-model.php [hunk 1]
```diff
@@ -288,33 +295,69 @@ public function update_language( $args ) {
...
+	 *   @type int    $lang_id    Optional, ID of the language to modify. An empty value means the language is being
+	 *                            created.
+	 *   @type string $flag       Optional, country code, {@see settings/flags.php}.
 	 * }
-	 * @return mixed The serialized description array updated.
+	 * @return string The serialized description array updated.
 	 */
-	protected function update_language_description( $args ) {
-		$language_term = get_term( (int) $args['lang_id'] );
+	protected function update_language_description( array $args ) {
```
Comment (Chouby) [inline admin/admin-model.php:310]: I like the name of your filter. Maybe we could rename this method the same way.

File: admin/admin-model.php [hunk 2]
```diff
@@ -288,33 +295,69 @@ public function update_language( $args ) {
...
+		 *                              not be loaded...
+		 *     @type int    $rtl        1 if rtl language, 0 otherwise.
+		 *     @type int    $term_group Language order when displayed.
+		 *     @type int    $lang_id    Optional, ID of the language to modify. An empty value means the language is
+		 *                              being created.
+		 *     @type string $flag       Optional, country code, {@see settings/flags.php}.
+		 * }
+		 */
+		$language_data = apply_filters( 'pll_language_metas', $language_data, $args );
+		// Don't allow to overwrite `$locale`, `$rtl`, and `$flag_code`.
```
Comment (Chouby) [inline admin/admin-model.php:355]: I agree with you not to allow to overwrite these properties. But having theme in the filter could be pertubating. Could we do something like
```php
$language_data = apply_filters( 'pll_language_metas', array(), $args );
```

File: admin/admin-model.php [hunk 3]
```diff
@@ -288,33 +295,69 @@ public function update_language( $args ) {
...
+		$new_data      = array(
+			'locale'    => $args['locale'],
+			'rtl'       => ! empty( $args['rtl'] ) ? 1 : 0,
+			'flag_code' => empty( $args['flag'] ) ? '' : $args['flag'],
 		);
+		$language_data = array_merge( $old_description, $new_data );
+
+		/**
+		 * Allow to add data to store for a language.
+		 *
```
Comment (Chouby) [inline admin/admin-model.php:332]: You forgot `@since`.

**Review by Chouby — CHANGES_REQUESTED** [review]

File: admin/admin-model.php (hunk 2 above)
Comment (Screenfeed) [reply admin/admin-model.php:355]: That's what I did first lol, but thought I was over-complicating things.

File: admin/admin-model.php (hunk 1 above)
Comment (Screenfeed) [reply admin/admin-model.php:310]: Renamed into `build_language_metas()` in [f3c72a8](https://github.com/polylang/polylang/pull/1201/commits/f3c72a86f125fdcfe952183a88f44803d6e6d8f2). I thought about `build_and_serialize_language_metas()` but it's a bit long 😅

File: admin/admin-model.php (hunk 2 above)
Comment (Screenfeed) [reply admin/admin-model.php]: Done in [f3c72a8](https://github.com/polylang/polylang/pull/1201/commits/f3c72a86f125fdcfe952183a88f44803d6e6d8f2).

**Review by Chouby — CHANGES_REQUESTED** [review]

File: admin/admin-model.php [hunk 4]
```diff
@@ -283,38 +290,78 @@ public function update_language( $args ) {
...
+		 * @since 3.4
+		 *
+		 * @param mixed[] $new_data      Data to add.
+		 * @param mixed[] $language_data {
+		 *     Original data being stored.
+		 *
+		 *     @type string $locale    WordPress locale.
+		 *     @type int    $rtl       1 if rtl language, 0 otherwise.
+		 *     @type string $flag_code Country code.
+		 * }
```
[the remark this answers is not in the document]
Comment (Screenfeed) [reply admin/admin-model.php]: Done in [efe562f](https://github.com/polylang/polylang/pull/1201/commits/efe562f5f9d8ae2310dae3f5bc0bc3180c6b162e).

**Review by Chouby — APPROVED** [review]

**Pull request merged**